### PR TITLE
initial tag archiving pages

### PR DESCRIPTION
**Pull request by maxandersen**
Here is the basic mechanics to create archive of tags.

uses jekyll-archives so can be used to archive anything but it of course slows down rendering of the site so for now just enabled for tags.

here is a screenshot of how it currently looks like:

gives you an idea, but need some ux TLC :)

### CODE REVIEW

**Comment by maxandersen** [comment]
fixes #448

**Comment by maxandersen** [comment]
updated with:

* commits to clean up current use of tags
* add section to readme on proposed tag usage
* added #'tag links to date header in overview and post page (as an example to make the tags pages available without too much extra scaffolding)

screenshots:

File: _layouts/tag-archive.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+layout: base
+---
+
+<h1>Archive of posts with {{ page.type }} '{{ page.title }}'</h1>
```
Comment (emmanuelbernard) [inline _layouts/tag-archive.html]:
```suggestion
<h1>Posts with {{ page.type }} '{{ page.title }}'</h1>
```

I don't like Archive as a name.

**Comment by emmanuelbernard** [comment]
@maxandersen the current branch has zero tag nor readme info. You did not push your latest work

**Comment by maxandersen** [comment]
> @maxandersen the current branch has zero tag nor readme info. You did not push your latest work

doh - pr was made on origin/tagarchive rather than maxandersen/tagarchive - pushed the latest.

I'll do an update more that remove the links so we can at least push the page to link to.

File: README.md [hunk 2]
```diff
@@ -39,10 +39,17 @@ These instructions will get you a copy of the Quarkus.io website up and running
 
 To write a blog:
 
-- create an author ent ry in [_data/authors.yaml](https://github.com/quarkusio/quarkusio.github.io/blob/develop/_data/authors.yaml)
+- create an author entry in [_data/authors.yaml](https://github.com/quarkusio/quarkusio.github.io/blob/develop/_data/authors.yaml)
     - `emailhash` is used to fetch your picture form the Gravatar service
 - create an blog entry under [_posts](https://github.com/quarkusio/quarkusio.github.io/tree/develop/_posts)
     -the file name is `yyyy-mm-dd-slug.adoc`
+- `tags` should be used with some care as an archive page is created for of them. Below are some basic rules to try follow:
+  - `quarkus-release` used for Quarkus release blogs
```
Comment (emmanuelbernard) [inline README.md]: I would prefer the simpler "release" I think.

File: README.md [hunk 3]
```diff
@@ -39,10 +39,17 @@ These instructions will get you a copy of the Quarkus.io website up and running
...
+  - `announcement` used for general announcement with some impact.
```
Comment (emmanuelbernard) [inline README.md]: Got examples?

**Comment by maxandersen** [comment]
done - cleanup tags, document tags, add `/tag/*/` urls.

no links to those pages from pages as still waiting for design team on feedback here.

File: README.md [hunk 4]
```diff
@@ -39,10 +39,17 @@ These instructions will get you a copy of the Quarkus.io website up and running
...
+  - `extension` used for blogs related to a specific extension.
+  - `user-story` used for stories from users/companies adopting Quarkus.
+  - `development-tips` used for blogs with tips to develop using Quarkus or Quarkus itself. 
+  - add a tech specific, like `kafka`, if your post has a signficant mention/relevance to that technology.
```
Comment (emmanuelbernard) [inline README.md]: this is an open door for many tags, so we should guide more here.

File: _posts/2019-11-21-vscode-quarkus-1.2.0.adoc [hunk 5]
```diff
@@ -2,6 +2,7 @@
 layout: post
 title: 'Quarkus Tools for Visual Studio Code - 1.2.0 release'
 date: 2019-11-21
+tags: ide vscode
```
Comment (emmanuelbernard) [inline _posts/2019-11-21-vscode-quarkus-1.2.0.adoc]: just ide no? vscode is like kafka-streams, details?

File: _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc [hunk 6]
```diff
@@ -2,7 +2,7 @@
 layout: post
 title: 'Sentry Error Tracker - A guardian of your Quarkus application'
 date: 2020-01-20
-tags: quarkus extension logging cloud error
+tags: extension sentry tracing
```
Comment (emmanuelbernard) [inline _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc]: sentry?

File: _posts/2020-01-24-quarkus-wins-devies-award.adoc [hunk 7]
```diff
@@ -2,7 +2,7 @@
 layout: post
 title: Quarkus wins DEVIES award
 date: 2020-01-24
-tags: quarkus awards
+tags: announcement awards
```
Comment (emmanuelbernard) [inline _posts/2020-01-24-quarkus-wins-devies-award.adoc]: are we expecting 50 blongs on awards and wanting to filter them out?

File: _posts/2020-01-28-vscode-quarkus-1.3.0.adoc [hunk 8]
```diff
@@ -3,6 +3,7 @@ layout: post
 title: 'Introducing 10 new features in Quarkus Tools for Visual Studio Code 1.3.0'
 date: 2020-02-07
 author: dakwon
+tags: ide vscode
```
Comment (emmanuelbernard) [inline _posts/2020-01-28-vscode-quarkus-1.3.0.adoc]: see above

File: _config.yml [hunk 9]
```diff
@@ -153,3 +154,19 @@ publication:
...
+jekyll-archives:
+  enabled: 
+    - tags
+  layouts: 
+    tag: tag-archive
+  permalinks:
+    year: '/:year/'
+    month: '/:year/:month/'
+    day: '/:year/:month/:day/'
+    tag: '/tag/:name/'
```
Comment (emmanuelbernard) [inline _config.yml]: I'd do `/blog/tag/:name/` or are we planning to tag non blog pages and have them visible across?

File: _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc (hunk 6 above)
Comment (maxandersen) [reply _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc]: applying the "tag for main technology the extension blog is about" ...

File: _posts/2019-11-21-vscode-quarkus-1.2.0.adoc (hunk 5 above)
Comment (maxandersen) [reply _posts/2019-11-21-vscode-quarkus-1.2.0.adoc]: we can, not married to it.

File: README.md (hunk 2 above)
Comment (maxandersen) [reply README.md]: just done to in future maybe have platform-release, etc. but fine just using `release` for main quarkus and if we start having others we just add them.

File: _posts/2020-01-24-quarkus-wins-devies-award.adoc (hunk 7 above)
Comment (maxandersen) [reply _posts/2020-01-24-quarkus-wins-devies-award.adoc]: well, when I did this there was discussions on what we do when we have more than one devies award; so I treated it as something interesting to have a page for like `user-story`

File: _posts/2020-01-28-vscode-quarkus-1.3.0.adoc (hunk 8 above)
Comment (maxandersen) [reply _posts/2020-01-28-vscode-quarkus-1.3.0.adoc]: ive removed vscode,eclipse,che,intellij tags.

File: _config.yml (hunk 9 above)
Comment (maxandersen) [reply _config.yml]: the jekyll-archives plugin just do it for posts so maybe best to just put under blog/tags/

File: _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc (hunk 6 above)
Comment (emmanuelbernard) [reply _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc]: I always come back to, will we want to filter blogs by sentry refs?

Comment (maxandersen) [reply _posts/2020-01-20-sentry-the-guardian-of-your-quarkus-app.adoc]: right now? probably not so much - over time..yeah...similar to "hibernate", "kafka", etc. wuold be interesting for those with tech specific interest.

**Comment by emmanuelbernard** [comment]
My experience with Awestruct is that more tags == exponential generation
time. So I would not take it lightly.

**Pull request merged**